### DISCUSSION TITLE
How do I fix this?

### DISCUSSION THREAD

**Discussion by Amaranth-101** [community/community#138051] [Programming Help] [answered]
### How do I fix this?

![IMG_0650](https://github.com/user-attachments/assets/31523dc8-4794-4022-b87d-5844a5dff0cd)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment]
Hi @Amaranth-101 ,

It looks like you haven't imported the `datetime` module, add this to the beginning of your code:

```python
from datetime import datetime
```

Take a look at the python documentation for more information on importing modules in python:

- [Modules](https://docs.python.org/3/tutorial/modules.html)

And further info about the `datetime` module:

- [datetime — Basic date and time types](https://docs.python.org/3/library/datetime.html)

**Reply by Amaranth-101** [reply]
I thought I had imported it 😅
Edit: I imported it but it still didn’t work for some reason.

**Reply by davevad93** [reply]
Hey @Amaranth-101 I see, could you paste here your code and the error it returns?

**Reply by Amaranth-101** [reply]
It says “NameError: name ‘datetime’ is not defined” on my terminal when I write “python Keylogger.py”. I installed datetime and imported it like you said, but I’ve done no progress.

**Reply by davevad93** [reply, marked as answer]
That's odd @Amaranth-101 because the datetime module has been imported, have you placed the import at the top of your script like this?

```python
from datetime import datetime

class Keylogger:
   def __init__(self, interval, report_method="email"): 
      # Rest of your code...
```
